Sming/SmingCore/Network/TcpConnection.h

```diff
@@ -228,6 +268,16 @@ class TcpConnection
 	virtual err_t onSslConnected(SSL* ssl);
 #endif
 
+	// These do most of the work - called from static methods
+	err_t tcpOnConnected(err_t err);
```

Comment:
Can you rename the `tcpOn*` to `interalOn*`? Otherwise having staticOnConnect, tcpOnConnect, onConnect will cause additional confusion as to which method is the right one to use.

While `internalOn*` might be a bit better than `tcpOn*` and not quite happy with it so I am also open to suggestions from your side.

Reply:
We could adopt the LWIP convention e.g. `tcp_connected` instead of `tcpOnConnect` to indicate it's all happening at a lower level. Could do the same with the static methods, e.g. `static_tcp_connected`...

Reply:
Just noticed, we already have an `internalTcpConnect`!

Reply:
This will come up again handling HTTP and Websocket parser callbacks. These are simpler though as the static methods are all virtually identical (just a null pointer check then call the method). Both parsers use a function table, so I adopted the same names as those to help make things clearer, e.g. `HttpConnection::on_message_begin`.

Reply:
> Just noticed, we already have an `internalTcpConnect`!

Nice catch :)

For consistency I can suggest the following method renames:
```
internalTcpConnect -> internalConnect
tcpOn* -> internalOn* (Ex: tcpOnConnected -> internalOnConnected )
```

Can you do that change? And don't forget to remove the commented line 20 in Sming/SmingCore/Network/MqttClient.h.